packages/cli/src/lib/snapshot/reportViewer.ts

```diff
@@ -0,0 +1,185 @@
+        `${row.operations.resourcesUpdated} to update`
+      )}\n`;
+    }
+    // TODO: CDX-361: Only show delete items if delete flag is set to true
+    if (row.operations.resourcesDeleted > 0) {
+      output += `${this.style.red('-')}   ${this.style.red(
+        `${row.operations.resourcesDeleted} to delete`
+      )}\n`;
+    }
+    if (row.operations.resourcesInError > 0) {
+      output += `${this.style.error(
+        `!   ${row.operations.resourcesInError} in error `
+      )}\n`;
+    }
+
+    return output;
+  }
+
+  private get changedResources() {
+    type resourceEntries = [string, ResourceSnapshotsReportOperationModel];
+    const resourceHasAtLeastOneOperation = ([
+      _,
+      operations,
+    ]: resourceEntries) => {
+      return (
+        operations.resourcesCreated +
+          operations.resourcesUpdated +
+          operations.resourcesRecreated +
+          // TODO: CDX-361: Only count delete items if delete flag is set to true
+          operations.resourcesDeleted +
+          operations.resourcesInError >
+        0
+      );
+    };
+
+    const convertArrayToObject = ([
+      resourceName,
+      operations,
+    ]: resourceEntries) => ({
+      resourceName,
+      operations,
+    });
+
+    return Object.entries(this.report.resourceOperations)
+      .filter(resourceHasAtLeastOneOperation)
+      .map(convertArrayToObject);
+  }
+
+  private prettyPrintResourceName(resourceName: string): string {
+    const capitalized =
+      resourceName.charAt(0) + resourceName.slice(1).toLowerCase() + 's';
+    return capitalized.replace(/_/g, ' ');
+  }
+
+  private isSuccessReport(): boolean {
+    return this.report.resultCode === ResourceSnapshotsReportResultCode.Success;
+  }
+
+  private getOperationTypeTotalCount(
+    type: keyof ResourceSnapshotsReportOperationModel
+  ) {
+    const count = Object.values(this.report.resourceOperations).reduce(
+      (count: number, current: ResourceSnapshotsReportOperationModel) =>
+        count + current[type],
+      0
+    );
+
+    return count;
+  }
+
+  private handleReportErrors() {
+    const maximumNumberOfErrorsToPrint = 5;
+    const totalErrorCount = this.getOperationTypeTotalCount('resourcesInError');
+
+    cli.log(this.style.header('Error Report:'));
+    cli.log(
+      this.style.error(
+        `   ${totalErrorCount} resource${
+          totalErrorCount > 1 ? 's' : ''
+        } in error `
+      )
+    );
+
+    for (const resourceType in this.report.resourceOperationResults) {
+      let remainingErrorsToPrint = maximumNumberOfErrorsToPrint;
+      const operationResult =
+        this.report.resourceOperationResults[resourceType];
+      const hasNoError = (op: ResourceSnapshotsReportOperationResults) =>
+        Object.keys(op).length === 0;
+
+      if (hasNoError(operationResult)) {
+        continue;
+      }
+
+      cli.log(`\n ${this.prettyPrintResourceName(resourceType)}`);
+
+      for (const resourceInError in operationResult) {
```

Comment:
Could this be simplified by splicing the array of errors ?

```
const toDisplay = operationResult.splice(0, maximumNumberOfErrorsToPrint);

// display each "toDisplay"
// display operationResult.length error left/unprinted
```

You can also clone operationResult first if you do not want to modify the original array.

Reply:
It's not that simple. `operationResult` is a record of record of array. There are 2 ways we can reach the maximum number of errors to print:

1. Having more than 5 entries within the same **resource type** (first-level record)
2. Having multiple errors within the same **resource** (second-level record)

That's why I am using nested for loops even if it's not pretty :S 

Here is an example:
```
  resourceOperationResults: {
    FIELD: {
      foo_4VNj5ds5: ['RESOURCE_ALREADY_EXISTS: Field foo already exists.'],
      bar_4VNj5ds5: ['RESOURCE_ALREADY_EXISTS: Field bar already exists.'],
      dsads_4VNj5ds5: ['RESOURCE_ALREADY_EXISTS: Field dsads already exists.'],
      fdww_4VNj5ds5: [
        'RESOURCE_ALREADY_EXISTS: Field fdww already exists.',
        'ERROR_CODE: another error',
        'ERROR_CODE: another another error',
        'ERROR_CODE: another another another error',
        'ERROR_CODE: another another another another error',
        'ERROR_CODE: another another another another another error',
      ],
      csad_4VNj5ds5: ['RESOURCE_ALREADY_EXISTS: Field csad already exists.'],
      hjkd_4VNj5ds5: ['RESOURCE_ALREADY_EXISTS: Field hjkd already exists.'],
      fdasf_4VNj5ds5: ['RESOURCE_ALREADY_EXISTS: Field fdasf already exists.'],
    },
  },
```

And it should print
```
* RESOURCE_ALREADY_EXISTS: Field foo already exists
* RESOURCE_ALREADY_EXISTS: Field bar already exists
* RESOURCE_ALREADY_EXISTS: Field dsads already exists
* RESOURCE_ALREADY_EXISTS: Field fdww already exists
* ERROR_CODE: another error
* ERROR_CODE: another another error
```